browser/ui/views/profiles/brave_profile_menu_view.cc

```diff
@@ -42,19 +43,28 @@ void BraveProfileMenuView::SetProfileIdentityInfo(
   // pointers being assigned to those members.
 
   Profile* profile = browser()->profile();
-  ProfileAttributesEntry* profile_attributes =
-      g_browser_process->profile_manager()
-          ->GetProfileAttributesStorage()
-          .GetProfileAttributesWithPath(profile->GetPath());
-  // Reset IdentityInfo to get rid of the subtitle string
-  // IDS_PROFILES_LOCAL_PROFILE_STATE("Not signed in").
-  ProfileMenuView::SetProfileIdentityInfo(
-      /*profile_name=*/std::u16string(),
-      profile_attributes->GetProfileThemeColors().profile_highlight_color,
-      /*edit_button_params=*/std::nullopt,
-      ui::ImageModel::FromImage(profile_attributes->GetAvatarIcon()),
-      ui::ImageModel(),
-      /*title=*/profile_attributes->GetName());
+  if (profile->IsGuestSession()) {
+    ProfileMenuView::SetProfileIdentityInfo(
+        /*profile_name=*/std::u16string(),
+        /*background_color=*/SK_ColorTRANSPARENT,
+        /*edit_button_params=*/std::nullopt, profiles::GetGuestAvatar(),
+        ui::ImageModel(),
+        /*title=*/l10n_util::GetStringUTF16(IDS_GUEST_PROFILE_NAME));
```

Comment:
How about passing received args directly?
I think using args from `BuildGuestIdentity()` here would be better.
```
    ProfileMenuView::SetProfileIdentityInfo(
        profile_name, profile_background_color, edit_button_params, image_model,
        management_badge, title, subtitle, management_label, header_art_icon);
```

Reply:
Oh wow, I completely missed that this function is passed all of those args! That will greatly simplify this, thanks for catching that.

Pushed a fix, @cdesouza-chromium may double-check me on this one that what I'm doing here is right?

Reply:
Hmm, I think @cdesouza-chromium replaced some of args to avoid dangling issue.
What I thought first was below but some of them also should be replaced to prevent dangling?
``` 
if (profile->IsGuestSession()) {
    ProfileMenuView::SetProfileIdentityInfo(
        profile_name, profile_background_color, edit_button_params, image_model,
        management_badge, title, subtitle, management_label, header_art_icon);
} else {
   ...
}
```

Reply:
hmmm, so my dangling pointer change introduced a crash here... would it be possible to have a small browser test covering this case (I don't want to ask too much if the test is not easy to write)

Reply:
Right, I see what you're saying now. Let me push the change you're suggesting and we'll see what the Linux CI thinks (that's where the dangling pointer detection is currently enabled). If it doesn't like it, then I can address that based on the debug info it provides.

Reply:
I'm pretty sure your change is fine when it comes to dangling pointer... the issue I originally fixed is that we were calling `SetProfileIdentityInfo` twice, which was causing an intermediate state during the execution of the second call, where a dangling pointer check was causing a crash.

Reply:
I think we probably want to reuse`subtitle` at least because it's dynamically generated based on guest window title.

Reply:
Right, I'll just go back to forwarding the args then since that doesn't affect the dangling pointer fix by the sound of it.

Reply:
If just pass all args to base class' method and it's ok now,  I think we could just delete this method override?

Reply:
OK, CI passed so should be all set to review again!